invisible_cities/reco/peak_functions.py

```diff
@@ -153,9 +153,10 @@ def rebin_times_and_waveforms(times, widths, waveforms,
     rebinned_wfs    = np.zeros((n_sensors, len(slices)))
 
     for i, s in enumerate(slices):
-        t  = times    [   s]
-        e  = waveforms[:, s]
-        w  = np.sum(e, axis=0) if np.any(e) else None
+        t = times    [   s]
+        e = waveforms[:, s]
+        w = np.sum(e, axis=0).clip(0) if np.any(e) else None
+        if not np.any(w): w = None
```

Comment:
This is equivalent to a more readable version (at least to me):
```
w = np.sum(e, axis=0).clip(0) #note that we dont have to check if there are non-0s in e  since we will check the value of the sum later
if np.all(w == 0) : w = None
```
Maybe this can be made a bit more elegant so we dont rewrite `w`